### PR TITLE
Upgrade to Patternfly 5

### PR DESCRIPTION
**Pull request by bipuladh**
Major changes: 
1) Packages related to Patternfly upgraded to v5.
2) Changes to how form validation is done.
......

### CODE REVIEW

File: packages/shared/src/status/icons.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import * as React from 'react';
 import { global_danger_color_100 as dangerColor } from '@patternfly/react-tokens/dist/js/global_danger_color_100';
-import { global_default_color_200 as blueDefaultColor } from '@patternfly/react-tokens/dist/js/global_default_color_200';
 import { global_disabled_color_100 as disabledColor } from '@patternfly/react-tokens/dist/js/global_disabled_color_100';
+import { global_info_color_100 as blueDefaultColor } from '@patternfly/react-tokens/dist/js/global_info_color_100';
 import { global_palette_blue_300 as blueInfoColor } from '@patternfly/react-tokens/dist/js/global_palette_blue_300';
 import { global_palette_green_500 as okColor } from '@patternfly/react-tokens/dist/js/global_palette_green_500';
 import { global_warning_color_100 as warningColor } from '@patternfly/react-tokens/dist/js/global_warning_color_100';
```
Comment (SanjalKatiyar) [inline packages/shared/src/status/icons.tsx:7]: same question here, regarding import from `dist` (https://github.com/openshift/console/pull/13637, `upgrade-PatternFly.md` file, section `Console 4.15`).

File: packages/shared/src/table/selectable-table.tsx [hunk 2]
```diff
@@ -1,10 +1,11 @@
 import * as React from 'react';
 import { StatusBox } from '@odf/shared/generic/status-box';
 import { K8sResourceCommon } from '@openshift-console/dynamic-plugin-sdk';
+import { Table } from '@patternfly/react-table/deprecated';
+import { TdSelectType } from '@patternfly/react-table/dist/esm/components/Table/base/types';
```
Comment (SanjalKatiyar) [inline packages/shared/src/table/selectable-table.tsx:5]: import from `dist`, instead of corresponding `index`...

File: tsconfig.json [hunk 3]
```diff
@@ -10,7 +10,7 @@
     "experimentalDecorators": true,
     "noUnusedLocals": true,
     "skipLibCheck": true,
-    "sourceMap": true,
+    "sourceMap": false,
```
Comment (SanjalKatiyar) [inline tsconfig.json:13]: is it possible to set this based on dev/prod env ?

File: webpack.config.ts [hunk 4]
```diff
@@ -113,13 +112,13 @@ const config: webpack.Configuration & DevServerConfiguration = {
           {
             loader: 'css-loader',
             options: {
-              sourceMap: true,
+              sourceMap: false,
```
Comment (SanjalKatiyar) [inline webpack.config.ts:115]: same for these, can we set this based on dev/prod env ??

File: webpack.config.ts [hunk 5]
```diff
@@ -158,25 +157,14 @@ const config: webpack.Configuration & DevServerConfiguration = {
     new webpack.ProvidePlugin({
       Buffer: ['buffer', 'Buffer'],
     }),
-    new ForkTsCheckerWebpackPlugin({
```
Comment (SanjalKatiyar) [inline webpack.config.ts:161]: we don't need it anymore ??

File: webpack.config.ts [hunk 6]
```diff
@@ -63,7 +62,7 @@ const config: webpack.Configuration & DevServerConfiguration = {
         include: /packages/,
         exclude: /(build|dist)/, // Ignore shared build folder.
         use: [
-          {
+          /*           {
```
Comment (SanjalKatiyar) [inline webpack.config.ts:65]: we don't need this as well ??

File: packages/mco/components/create-dr-policy/create-dr-policy.tsx [hunk 7]
```diff
@@ -201,29 +201,37 @@ const CreateDRPolicy: React.FC<{}> = () => {
...
+          <FormHelperText>
+            <HelperText>
+              <HelperTextItem>
+                {t(
+                  'Enables mirroring/replication between two selected clusters, ensuring failover or relocation between the two clusters in the event of an outage or planned maintenance.'
+                )}
+              </HelperTextItem>
+            </HelperText>
+          </FormHelperText>
           <SelectClusterList
```
Comment (GowthamShanmugam) [inline packages/mco/components/create-dr-policy/create-dr-policy.tsx:218]: Just for helper text do we need 3 elements in PF5? is helperText removed from formGroup?

File: packages/mco/components/modals/app-manage-policies/helper/pvc-details-wizard-content.tsx [hunk 8]
```diff
@@ -151,79 +143,69 @@ const PairElement: React.FC<PairElementProps> = ({
   }, [index, onRemoveProp]);
 
   return (
-    <Grid hasGutter>
-      <GridItem lg={5} sm={5}>
-        <FormGroup
-          hasNoPaddingTop
-          isRequired
+    <div className="row" data-test="pairs-list-row">
```
Comment (GowthamShanmugam) [inline packages/mco/components/modals/app-manage-policies/helper/pvc-details-wizard-content.tsx:146]: Upto 4.15 i used to keep the same logic, but it is started breaking after the recent change in 4.15, i recently switch to grid to solve this problem:https://github.com/red-hat-storage/odf-console/pull/1235

File: packages/ibm/system-connection-details.tsx [hunk 9]
```diff
@@ -12,14 +12,14 @@ import { useCustomTranslation } from '@odf/shared/useCustomTranslationHook';
 import { isValidIP } from '@odf/shared/utils';
 import { getAPIVersionForModel } from '@odf/shared/utils';
 import { k8sGet } from '@openshift-console/dynamic-plugin-sdk';
+import { Select, SelectOption } from '@patternfly/react-core/deprecated';
```
Comment (GowthamShanmugam) [inline packages/ibm/system-connection-details.tsx:15]: import SelectOption from '@patternfly/react-core'

File: packages/ocs/storage-class/sc-form.tsx [hunk 10]
```diff
@@ -460,18 +462,14 @@ export const StorageClassEncryption: React.FC<ProvisionerProps> = ({
   return (
     <div className="ocs-storage-class__form">
       <Form>
-        <FormGroup
-          fieldId="storage-class-encryption"
-          helperTextInvalid={t('This is a required field')}
```
Comment (GowthamShanmugam) [inline packages/ocs/storage-class/sc-form.tsx:465]: Ignoring helperTextInvalid?

File: packages/odf/components/kms-config/hpcs-config.tsx [hunk 11]
```diff
@@ -62,99 +69,144 @@ export const HpcsConfigure: React.FC<KMSConfigureProps> = ({
...
           name="kms-service-name"
           isRequired
           validated={isValid(kms.name.valid)}
           data-test="kms-service-name-text"
         />
+
+        <FormHelperText>
+          <HelperText>
+            <HelperTextItem
+              variant={isValid(kms.name.valid) ? 'default' : 'error'}
```
Comment (GowthamShanmugam) [inline packages/odf/components/kms-config/hpcs-config.tsx:88]:
```suggestion
              variant={getValidatedProp(!kms.name.valid)}
```

Comment (GowthamShanmugam) [inline packages/odf/components/kms-config/hpcs-config.tsx:88]:
```suggestion
              variant={getValidatedProp(!kms.name.valid)}
```

File: packages/odf/components/kms-config/hpcs-config.tsx [hunk 12]
```diff
@@ -62,99 +69,144 @@ export const HpcsConfigure: React.FC<KMSConfigureProps> = ({
...
+            >
+              {isValid(kms.name.valid) === ValidatedOptions.default
```
Comment (GowthamShanmugam) [inline packages/odf/components/kms-config/hpcs-config.tsx:90]:
```suggestion
              {getValidatedProp(!kms.name.valid) === ValidatedOptions.default
```

File: packages/odf/components/kms-config/hpcs-config.tsx (hunk 11 above)
Comment (GowthamShanmugam) [reply packages/odf/components/kms-config/hpcs-config.tsx:88]: Optional changes

File: packages/odf/components/kms-config/hpcs-config.tsx [hunk 13]
```diff
@@ -62,99 +69,144 @@ export const HpcsConfigure: React.FC<KMSConfigureProps> = ({
...
           name="kms-instance-id"
           isRequired
           validated={isValid(kms.instanceId.valid)}
           data-test="kms-instance-id-text"
         />
+        <FormHelperText>
+          <HelperText>
+            {isValid(kms.instanceId.valid) === ValidatedOptions.error && (
+              <HelperTextItem
+                variant={isValid(kms.instanceId.valid) ? 'default' : 'error'}
```
Comment (GowthamShanmugam) [inline packages/odf/components/kms-config/hpcs-config.tsx:119]:
```suggestion
                variant={getValidatedProp(!kms.instanceId.valid)}
```
Optional change

File: packages/odf/components/kms-config/name-address-port.tsx [hunk 14]
```diff
@@ -63,23 +70,31 @@ export const NameAddrPort: React.FC<NameAddrPortProps> = ({
...
           id="kms-service-name"
           name="kms-service-name"
           isRequired
           validated={isValid(kmsState.name.valid)}
           data-test="kms-service-name-text"
         />
+        <FormHelperText>
+          <HelperText>
+            <HelperTextItem
+              variant={isValid(kmsState.name.valid) ? 'default' : 'error'}
```
Comment (GowthamShanmugam) [inline packages/odf/components/kms-config/name-address-port.tsx:88]:
```suggestion
              variant={getValidatedProp(!kmsState.name.valid)}
```

I think isValid itself returns 'default' | 'error'

File: packages/odf/components/kms-config/name-address-port.tsx [hunk 15]
```diff
@@ -104,6 +117,16 @@ export const NameAddrPort: React.FC<NameAddrPortProps> = ({
             validated={isValid(kmsState.address.valid)}
             data-test="kms-address-text"
           />
+          <FormHelperText>
+            <HelperText>
+              <HelperTextItem
+                variant={isValid(kmsState.address.valid) ? 'default' : 'error'}
```
Comment (GowthamShanmugam) [inline packages/odf/components/kms-config/name-address-port.tsx:123]:
```suggestion
                variant={getValidatedProp(!kmsState.address.valid) }
```

File: packages/odf/components/kms-config/name-address-port.tsx [hunk 16]
```diff
@@ -104,6 +117,16 @@ export const NameAddrPort: React.FC<NameAddrPortProps> = ({
...
+              >
+                {isValid(kmsState.address.valid) === ValidatedOptions.error &&
```
Comment (GowthamShanmugam) [inline packages/odf/components/kms-config/name-address-port.tsx:125]:
```suggestion
                {getValidatedProp(!kmsState.address.valid) === ValidatedOptions.error &&
```

File: packages/odf/components/kms-config/name-address-port.tsx [hunk 17]
```diff
@@ -112,20 +135,28 @@ export const NameAddrPort: React.FC<NameAddrPortProps> = ({
...
             id="kms-address-port"
             name="kms-address-port"
             isRequired
             validated={isValid(kmsState.port.valid)}
             data-test="kms-address-port-text"
           />
+          <FormHelperText>
+            <HelperText>
+              <HelperTextItem
+                variant={isValid(kmsState.port.valid) ? 'default' : 'error'}
```
Comment (GowthamShanmugam) [inline packages/odf/components/kms-config/name-address-port.tsx:153]:
```suggestion
                variant={getValidatedProp(!kmsState.port.valid)}
```
or just isValid(kmsState.port.valid)  is enough, isValid already returns  'default' : 'error'

File: packages/odf/components/kms-config/thales-config.tsx [hunk 18]
```diff
@@ -88,25 +96,35 @@ const FileUploadInput: React.FC<FileUploadInputProps> = ({
...
+          },
           maxSize: KMSMaxFileUploadSize,
           onDropRejected: () => setError(KMSFileSizeErrorMsg),
         }}
         data-test={id}
       />
+      <FormHelperText>
+        <HelperText>
+          <HelperTextItem
+            variant={isValid(!thalesState[id].error) ? 'default' : 'error'}
```
Comment (GowthamShanmugam) [inline packages/odf/components/kms-config/thales-config.tsx:121]:
```suggestion
            variant={ isValid(!thalesState[id].error) }
```

File: packages/mco/components/empty-state-page/empty-page.tsx [hunk 19]
```diff
@@ -34,35 +35,38 @@ const EmptyPage: React.FC<EmptyPageProps> = (props) => {
...
   ) : (
     <div className="mco-empty-page__background">
-      <EmptyState variant={EmptyStateVariant.large}>
-        <EmptyStateIcon icon={EmptyPageIcon} />
-        <Title headingLevel="h4" size="lg">
-          {title}
-        </Title>
+      <EmptyState variant={EmptyStateVariant.lg}>
+        <EmptyStateHeader
+          titleText={<>{title}</>}
```
[the remark this answers is not in the document]
Comment (bipuladh) [reply packages/mco/components/empty-state-page/empty-page.tsx]: yes

File: packages/mco/components/modals/app-data-policies-status/subscriptions/dr-status-table.tsx [hunk 20]
```diff
@@ -17,7 +17,7 @@ import {
   Tooltip,
 } from '@patternfly/react-core';
 import {
-  TableComposable,
+  Table /* data-codemods */,
```
[the remark this answers is not in the document]
Comment (bipuladh) [reply packages/mco/components/modals/app-data-policies-status/subscriptions/dr-status-table.tsx:20]: removed

File: packages/mco/components/modals/app-manage-policies/policy-list-view.tsx [hunk 21]
```diff
@@ -208,13 +208,19 @@ export const PolicyListView: React.FC<PolicyListViewProps> = ({
...
-            variant={EmptyStateVariant.large}
+            variant={EmptyStateVariant.lg}
             className="mco-manage-policies__emptyState---margin-bottom"
           >
-            <EmptyStateIcon icon={InfoCircleIcon} color={blueInfoColor.value} />
-            <Title headingLevel="h3" size="lg">
-              {t('No assigned data policy found')}
-            </Title>
+            <EmptyStateHeader
+              titleText={<>{t('No assigned data policy found')}</>}
```
[the remark this answers is not in the document]
Comment (bipuladh) [reply packages/mco/components/modals/app-manage-policies/policy-list-view.tsx:215]: Expects React.ReactNode

File: packages/odf/components/bucket-class/backingstore-table.tsx [hunk 22]
```diff
@@ -260,7 +260,6 @@ const BackingStoreSelection: React.FC<BackingStoreSelectionProps> = (props) => {
             title={t(
               'Each BackingStore can be used for one tier at a time. Selecting a BackingStore in one tier will remove the resource from the second tier option and vice versa.'
             )}
-            aria-label={t("Bucket created for Data Foundation's Service")}
```
[the remark this answers is not in the document]
Comment (bipuladh) [reply packages/odf/components/bucket-class/backingstore-table.tsx:263]: oops! mistake, reverting

File: packages/odf/components/create-storage-system/create-storage-system-steps/capacity-and-nodes-step/capacity-and-nodes-step.tsx [hunk 23]
```diff
@@ -339,9 +339,9 @@ const SelectedCapacityAndNodes: React.FC<SelectedCapacityAndNodesProps> = ({
           <Grid hasGutter>
             <GridItem span={5}>
               <TextInput
-                isReadOnly
                 value={humanizeBinaryBytes(capacity).string}
                 id="available-raw-capacity"
+                readOnlyVariant="default"
```
[the remark this answers is not in the document]
Comment (bipuladh) [reply packages/odf/components/create-storage-system/create-storage-system-steps/capacity-and-nodes-step/capacity-and-nodes-step.tsx:344]: no values are hardcoded strings

File: packages/odf/components/kms-config/thales-config.tsx [hunk 24]
```diff
@@ -187,41 +205,58 @@ export const ThalesConfigure: React.FC<KMSConfigureProps> = ({
...
             name="kms-unique-id"
             isRequired
             validated={isValid(thalesState.uniqueId.valid)}
             data-test="kms-unique-id"
           />
+          <FormHelperText>
+            <HelperText>
+              <HelperTextItem
+                variant={
+                  isValid(thalesState.uniqueId.valid) ? 'default' : 'error'
```
Comment (GowthamShanmugam) [inline packages/odf/components/kms-config/thales-config.tsx:224]:
```suggestion
                  isValid(thalesState.uniqueId.valid) 
```

File: packages/odf/components/kms-config/hpcs-config.tsx [hunk 25]
```diff
@@ -62,99 +69,144 @@ export const HpcsConfigure: React.FC<KMSConfigureProps> = ({
...
-        validated={isValid(kms.name.valid)}
-        helperText={t(
-          'An unique name for the key management service within the project. Name must only include alphanumeric characters, "-", "_" or "."'
-        )}
         isRequired
       >
         <TextInput
           value={kms.name.value}
-          onChange={setParams(HPCSParams.NAME)}
+          onChange={(ev, value) => setParams(HPCSParams.NAME)(value)}
```
[the remark this answers is not in the document]
Comment (bipuladh) [reply packages/odf/components/kms-config/hpcs-config.tsx:76]: ack

File: packages/odf/components/kms-config/hpcs-config.tsx [hunk 26]
```diff
@@ -62,99 +69,144 @@ export const HpcsConfigure: React.FC<KMSConfigureProps> = ({
...
         label={t('Service instance ID')}
         className={`${className}__form-body`}
-        helperTextInvalid={t('This is a required field')}
-        validated={isValid(kms.instanceId.valid)}
         isRequired
       >
         <TextInput
           value={kms.instanceId.value}
-          onChange={setParams(HPCSParams.INSTANCE_ID)}
+          onChange={(ev, value) => setParams(HPCSParams.INSTANCE_ID)(value)}
```
[the remark this answers is not in the document]
Comment (bipuladh) [reply packages/odf/components/kms-config/hpcs-config.tsx:107]: ack

File: packages/odf/components/kms-config/hpcs-config.tsx [hunk 27]
```diff
@@ -62,99 +69,144 @@ export const HpcsConfigure: React.FC<KMSConfigureProps> = ({
...
         label={t('Service API key')}
         className={`${className}__form-body`}
-        helperTextInvalid={t('This is a required field')}
-        validated={isValid(kms.apiKey.valid)}
         isRequired
       >
         <TextInput
           value={kms.apiKey.value}
-          onChange={setParams(HPCSParams.API_KEY)}
+          onChange={(ev, value) => setParams(HPCSParams.API_KEY)(value)}
```
[the remark this answers is not in the document]
Comment (bipuladh) [reply packages/odf/components/kms-config/hpcs-config.tsx:135]: ack

File: packages/odf/components/kms-config/hpcs-config.tsx [hunk 28]
```diff
@@ -62,99 +69,144 @@ export const HpcsConfigure: React.FC<KMSConfigureProps> = ({
...
         label={t('Customer root key')}
         className={`${className}__form-body`}
-        helperTextInvalid={t('This is a required field')}
-        validated={isValid(kms.rootKey.valid)}
         isRequired
       >
         <TextInput
           value={kms.rootKey.value}
-          onChange={setParams(HPCSParams.ROOT_KEY)}
+          onChange={(ev, value) => setParams(HPCSParams.ROOT_KEY)(value)}
```
[the remark this answers is not in the document]
Comment (bipuladh) [reply packages/odf/components/kms-config/hpcs-config.tsx:163]: ack

File: packages/odf/components/kms-config/hpcs-config.tsx [hunk 29]
```diff
@@ -62,99 +69,144 @@ export const HpcsConfigure: React.FC<KMSConfigureProps> = ({
...
         label={t('IBM base URL')}
         className={`${className}__form-body`}
-        helperTextInvalid={t('This is a required field')}
-        validated={isValid(kms.baseUrl.valid)}
         isRequired
       >
         <TextInput
           value={kms.baseUrl.value}
-          onChange={setParams(HPCSParams.BASE_URL)}
+          onChange={(ev, value) => setParams(HPCSParams.BASE_URL)(value)}
```
[the remark this answers is not in the document]
Comment (bipuladh) [reply packages/odf/components/kms-config/hpcs-config.tsx:191]: ack

File: packages/odf/components/kms-config/name-address-port.tsx (hunk 14 above)
Comment (GowthamShanmugam) [inline packages/odf/components/kms-config/name-address-port.tsx:88]:
```suggestion
              variant={isValid(kmsState.name.valid)}
```

File: packages/odf/components/mcg-endpoints/s3-endpoint-type.tsx [hunk 30]
```diff
@@ -118,9 +123,9 @@ export const S3EndPointType: React.FC<S3EndpointTypeProps> = (props) => {
           render={({ value, onChange, onBlur }) => (
             <TextInput
               data-test={`${type.toLowerCase()}-s3-endpoint`}
-              onChange={(e) => {
+              onChange={(e, val) => {
```
[the remark this answers is not in the document]
Comment (bipuladh) [reply packages/odf/components/mcg-endpoints/s3-endpoint-type.tsx:126]: e is used here

File: packages/odf/components/kms-config/hpcs-config.tsx (hunk 13 above)
Comment (GowthamShanmugam) [inline packages/odf/components/kms-config/hpcs-config.tsx:119]:
```suggestion
                variant={isValid(kms.instanceId.valid)}
```

File: packages/odf/components/mcg-endpoints/s3-endpoint-type.tsx [hunk 31]
```diff
@@ -180,21 +189,25 @@ export const S3EndPointType: React.FC<S3EndpointTypeProps> = (props) => {
...
-                  {t('Switch to Secret')}
-                </Button>
+              <InputGroup translate={undefined}>
+                <InputGroupItem isFill>
+                  <TextInput
+                    id="access-key"
+                    data-test={`${type.toLowerCase()}-access-key`}
+                    value={value}
+                    onChange={(e, val) => {
+                      onChange(e);
```
[the remark this answers is not in the document]
Comment (bipuladh) [reply packages/odf/components/mcg-endpoints/s3-endpoint-type.tsx:199]: onChange expects an event object

File: packages/odf/components/mcg-endpoints/s3-endpoint-type.tsx [hunk 32]
```diff
@@ -212,9 +225,9 @@ export const S3EndPointType: React.FC<S3EndpointTypeProps> = (props) => {
                 value={value}
                 id="secret-key"
                 data-test={`${type.toLowerCase()}-secret-key`}
-                onChange={(e) => {
+                onChange={(e, val) => {
                   onChange(e);
```
[the remark this answers is not in the document]
Comment (bipuladh) [reply packages/odf/components/mcg-endpoints/s3-endpoint-type.tsx:229]: previously there was no value being given by onChange now it passes value as well so utilizing it.

File: packages/odf/components/kms-config/hpcs-config.tsx (hunk 11 above)
Comment (GowthamShanmugam) [inline packages/odf/components/kms-config/hpcs-config.tsx:88]:
```suggestion
              variant={isValid(kms.name.valid) }
```

File: packages/odf/components/kms-config/hpcs-config.tsx [hunk 33]
```diff
@@ -62,99 +69,144 @@ export const HpcsConfigure: React.FC<KMSConfigureProps> = ({
...
           name="kms-root-key"
           isRequired
           validated={isValid(kms.rootKey.valid)}
           data-test="kms-root-key-text"
         />
+        <FormHelperText>
+          <HelperText>
+            {isValid(kms.rootKey.valid) === ValidatedOptions.error && (
+              <HelperTextItem
+                variant={isValid(kms.rootKey.valid) ? 'default' : 'error'}
```
Comment (GowthamShanmugam) [inline packages/odf/components/kms-config/hpcs-config.tsx:175]:
```suggestion
                variant={isValid(kms.rootKey.valid)}
```

File: packages/odf/components/kms-config/hpcs-config.tsx [hunk 34]
```diff
@@ -62,99 +69,144 @@ export const HpcsConfigure: React.FC<KMSConfigureProps> = ({
...
           name="kms-api-key"
           isRequired
           validated={isValid(kms.apiKey.valid)}
           data-test="kms-api-key-text"
         />
+        <FormHelperText>
+          <HelperText>
+            {isValid(kms.apiKey.valid) === ValidatedOptions.error && (
+              <HelperTextItem
+                variant={isValid(kms.apiKey.valid) ? 'default' : 'error'}
```
Comment (GowthamShanmugam) [inline packages/odf/components/kms-config/hpcs-config.tsx:147]:
```suggestion
                variant={isValid(kms.apiKey.valid) }
```

File: packages/odf/components/kms-config/hpcs-config.tsx [hunk 35]
```diff
@@ -62,99 +69,144 @@ export const HpcsConfigure: React.FC<KMSConfigureProps> = ({
...
           name="kms-base-url"
           isRequired
           validated={isValid(kms.baseUrl.valid)}
           data-test="kms-base-url"
         />
+        <FormHelperText>
+          <HelperText>
+            {isValid(kms.baseUrl.valid) === ValidatedOptions.error && (
+              <HelperTextItem
+                variant={isValid(kms.baseUrl.valid) ? 'default' : 'error'}
```
Comment (GowthamShanmugam) [inline packages/odf/components/kms-config/hpcs-config.tsx:203]:
```suggestion
                variant={isValid(kms.baseUrl.valid)}
```

File: packages/odf/components/kms-config/vault-auth-methods.tsx [hunk 36]
```diff
@@ -24,43 +28,53 @@ export const VaultTokenConfigure: React.FC<VaultAuthMethodProps> = ({
...
+            >
+              {revealToken ? <EyeSlashIcon /> : <EyeIcon />}
+            </Button>
+          </Tooltip>
+        </InputGroupItem>
       </InputGroup>
+      <FormHelperText>
+        <HelperText>
+          <HelperTextItem
+            variant={isValid(vaultState.authValue.valid) ? 'default' : 'error'}
```
Comment (GowthamShanmugam) [inline packages/odf/components/kms-config/vault-auth-methods.tsx:67]:
```suggestion
            variant={isValid(vaultState.authValue.valid) }
```

File: packages/odf/components/kms-config/vault-auth-methods.tsx [hunk 37]
```diff
@@ -90,6 +102,16 @@ export const VaultServiceAccountConfigure: React.FC<VaultAuthMethodProps> = ({
         isRequired
         validated={isValid(vaultState.authValue?.valid)}
       />
+      <FormHelperText>
+        <HelperText>
+          <HelperTextItem
+            variant={isValid(vaultState.authValue?.valid) ? 'default' : 'error'}
```
Comment (GowthamShanmugam) [inline packages/odf/components/kms-config/vault-auth-methods.tsx:108]:
```suggestion
            variant={isValid(vaultState.authValue?.valid) }
```

File: packages/odf/components/kms-config/vault-config.tsx [hunk 38]
```diff
@@ -157,6 +161,16 @@ export const VaultConfigure: React.FC<KMSConfigureProps> = ({
             />
           ))}
         </FormSelect>
+        <FormHelperText>
+          <HelperText>
+            <HelperTextItem
+              variant={isValid(!!vaultState.authMethod) ? 'default' : 'error'}
```
Comment (GowthamShanmugam) [inline packages/odf/components/kms-config/vault-config.tsx:167]:
```suggestion
              variant={isValid(!!vaultState.authMethod)}
```

File: packages/shared/src/charts/tooltip.tsx [hunk 39]
```diff
@@ -1,22 +1,24 @@
 // Mostly just a copy-paste from patternfly.
 import * as React from 'react';
+import { ChartLegendTooltipStyles } from '@patternfly/react-charts/dist/js/components/ChartTheme/ChartStyles';
+import {
+  getLegendTooltipSize,
+  getLegendTooltipDataProps,
+  getLegendTooltipVisibleData,
+  getLegendTooltipVisibleText,
+} from '@patternfly/react-charts/dist/js/components/ChartUtils/chart-tooltip';
```
[the remark this answers is not in the document]
Comment (bipuladh) [reply packages/shared/src/charts/tooltip.tsx:9]: got no option as it fails TS checks without this

File: packages/shared/src/status/icons.tsx (hunk 1 above)
Comment (bipuladh) [reply packages/shared/src/status/icons.tsx:7]: fails without this

File: tsconfig.json (hunk 3 above)
Comment (bipuladh) [reply tsconfig.json]: not required anymore

File: webpack.config.ts (hunk 5 above)
Comment (bipuladh) [reply webpack.config.ts]: we need this

File: webpack.config.ts (hunk 6 above)
Comment (bipuladh) [reply webpack.config.ts]: everything is breaking with thread-loader. removing for now will create an issue for further investigation

File: packages/mco/components/create-dr-policy/create-dr-policy.tsx (hunk 7 above)
Comment (bipuladh) [reply packages/mco/components/create-dr-policy/create-dr-policy.tsx:218]: yes life is sad now.

File: packages/shared/src/dropdown/multiselectdropdown.tsx [hunk 40]
```diff
@@ -43,7 +43,7 @@ export const MultiSelectDropdown: React.FC<MultiSelectDropdownProps> = ({
     <Select
       variant={variant || SelectVariant.typeaheadMulti}
       aria-label={t('Select input')}
-      onToggle={setOpen}
+      onToggle={() => setOpen((o) => !o)}
```
Comment (GowthamShanmugam) [inline packages/shared/src/dropdown/multiselectdropdown.tsx:46]: what is o? options?

File: packages/ocs/storage-class/sc-form.tsx (hunk 10 above)
Comment (bipuladh) [reply packages/ocs/storage-class/sc-form.tsx:465]: added a comment to add it in future as bug fix

File: packages/shared/src/status/icons.tsx (hunk 1 above)
Comment (SanjalKatiyar) [reply packages/shared/src/status/icons.tsx:7]: ack !

File: analyzeTest.ts [hunk 41]
```diff
@@ -1,7 +1,7 @@
 import * as fs from 'fs';
 
 const pluginName = process.env.PLUGIN;
-const MAX_ASSET_SIZE = 3; //3 MiB
+const MAX_ASSET_SIZE = 17; //17 MiB
```
Comment (SanjalKatiyar) [inline analyzeTest.ts:4]: is this intentional ? or were you testing something ?

Comment (bipuladh) [reply analyzeTest.ts:4]: Yes I will dicuss with ocp team later to reduce bundle size

File: packages/ocs/storage-class/sc-form.tsx (hunk 10 above)
Comment (GowthamShanmugam) [reply packages/ocs/storage-class/sc-form.tsx:465]: ack

**Comment by SanjalKatiyar** [comment]
> /override ci/prow/odf-console-e2e-aws

need to merge this in order to send/test followup CI fixes on `release` side...

File: packages/mco/components/modals/protected-applications/managed-applications-modal.tsx [hunk 42]
```diff
@@ -80,7 +81,9 @@ const ManagedApplicationsModal: React.FC<CommonModalProps> = (props) => {
           isInline
         >
           {t('Continue to Applications page')}
-          <ArrowRightIcon size={IconSize.sm} className="pf-u-ml-sm" />
+          <Icon size="sm">
+            <ArrowRightIcon className="pf-u-ml-sm" />
```
Comment (SanjalKatiyar) [inline packages/mco/components/modals/protected-applications/managed-applications-modal.tsx:85]: there are other references in this file too...
```suggestion
            <ArrowRightIcon className="pf-v5-u-ml-sm" />
```

File: packages/mco/components/protected-applications/components.tsx [hunk 43]
```diff
@@ -137,10 +136,7 @@ export const EnrollApplicationButton: React.FC<{ isNoDataMessage?: boolean }> =
                 'pf-u-mt-md': isNoDataMessage,
               })}
             >
-              <OutlinedQuestionCircleIcon
-                size={IconSize.sm}
-                className="pf-u-mr-sm"
-              />
+              <OutlinedQuestionCircleIcon className="pf-u-mr-sm" />
```
Comment (SanjalKatiyar) [inline packages/mco/components/protected-applications/components.tsx:139]: there are other references in this file too...
```suggestion
              <OutlinedQuestionCircleIcon className="pf-v5-u-mr-sm" />
```

File: packages/mco/components/protected-applications/list-page.tsx [hunk 44]
```diff
@@ -202,6 +204,7 @@ const ProtectedAppsTableRow: React.FC<
         <Td translate={null} isActionCell>
           <ActionsColumn
             items={getRowActions(t, launcher, navigate, application)}
+            translate={null}
```
Comment (SanjalKatiyar) [inline packages/mco/components/protected-applications/list-page.tsx:207]: this file as well, contains references to PF4 classes...

File: packages/shared/src/badges/Badge.scss [hunk 45]
```diff
@@ -1,10 +1,10 @@
 .ocs-preview-badge {
   &.pf-c-label {
```
Comment (SanjalKatiyar) [inline packages/shared/src/badges/Badge.scss:2]: now we will have to override PF5 classes, I guess...
```suggestion
  &.pf-v5-c-label {
```

File: packages/shared/src/badges/Badge.scss [hunk 46]
```diff
@@ -1,10 +1,10 @@
...
     --pf-c-label--BackgroundColor: #d93f00;
-    --pf-c-label--BorderRadius: var(--pf-global--BorderRadius--sm);
+    --pf-c-label--BorderRadius: var(--pf-v5-global--BorderRadius--sm);
```
Comment (SanjalKatiyar) [inline packages/shared/src/badges/Badge.scss:4]: aren't these variable names gonna get changed too ??
```suggestion
    --pf-v5-c-label--BorderRadius: var(--pf-v5-global--BorderRadius--sm);
```

File: packages/shared/src/charts/tooltip.tsx (hunk 39 above)
[the remark this answers is not in the document]
Comment (SanjalKatiyar) [reply packages/shared/src/charts/tooltip.tsx:9]: okay...

File: packages/shared/src/table/composable-table.tsx [hunk 47]
```diff
@@ -63,11 +63,7 @@ export const ComposableTable: ComposableTableProps = <
       NoDataEmptyMsg={noDataMsg}
       skeleton={<div className="loading-skeleton--table pf-u-mt-lg" />}
```
Comment (SanjalKatiyar) [inline packages/shared/src/table/composable-table.tsx:64]: PF4 refs in this file...

File: packages/shared/src/table/selectable-table.tsx [hunk 48]
```diff
@@ -101,7 +102,7 @@ export const SelectableTable: SelectableTableProps = <
       data={sortedRows}
       skeleton={<div className="loading-skeleton--table pf-u-mt-lg" />}
```
Comment (SanjalKatiyar) [inline packages/shared/src/table/selectable-table.tsx:103]: PF4 refs...

**Pull request merged**